### DISCUSSION TITLE
利用 Github Actions 将简悦的每日回顾推送到 Telegram 或飞书群

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#3383] [服务]
> 每晚 19点40分，通过 Github Actions 与简悦 API，将每日回顾推送到你的 Telegram 或飞书群。

![](https://cdn.jsdelivr.net/gh/23784148/upload-images@main/simpered/notice/Github%20Actions.png)

# 效果

- 发送到 Telegram

  ![image-20220208120301218](https://cdn.jsdelivr.net/gh/23784148/upload-images@main/typora/20220208_1644292981.png)

- 发送到飞书群

  ![image-20220208115548111](https://cdn.jsdelivr.net/gh/23784148/upload-images@main/typora/20220208_1644292548.png)

# Github Actions

一个持续集成的自动化方案，具有很多玩法，春节期间抽时间用了用，所以诞生了这个项目，这个项目的特点是：

- 不需要编写任何代码
- 不需要任何额外服务支持

# 前置知识

- 需要配置 [知识库](https://www.yuque.com/kenshin/simpread/lglfy2)、 [资料库](https://www.yuque.com/kenshin/simpread/vcmg4o) 与 [开放平台](https://www.yuque.com/kenshin/simpread/ieu1w6)
- 需要使用 [简悦 · 同步助手 小二](https://t.me/simpread_bot) 或 [飞书群 以及 飞书 bot](https://github.com/Kenshin/simpread/discussions/3104)
- 需要注册 Github 账户

# 项目地址

https://github.com/Kenshin/simpread-actions

# 配置

- 得到你的 Telegram User Name，打开 Telegram 进入 Setting → Editor Profile

  ![image-20220208111736548](https://cdn.jsdelivr.net/gh/23784148/upload-images@main/typora/20220208_1644290256.png)

- 获得你的 Chat ID，加 [@username_to_id_bot](https://t.me/username_to_id_bot) 然后把你的 Telegram User Name 发给它，就能得到 Chat ID

  ![image-20220208111524646](https://cdn.jsdelivr.net/gh/23784148/upload-images@main/typora/20220208_1644290124.png)

- 登录到你的 Github 然后进入 https://github.com/Kenshin/simpread-actions 并 **Fork**

  ![image-20220208110720621](https://cdn.jsdelivr.net/gh/23784148/upload-images@main/typora/20220208_1644289640.png)

- Fork 后设置 Actions secrets，位置在 Settings → Secrets → Actions，并新建 **New repository secret**

  ![image-20220208110937771](https://cdn.jsdelivr.net/gh/23784148/upload-images@main/typora/20220208_1644289777.png)

  > 分别建立以下几个 Secrets：

  - `SEND_TYPE` → 选择发送的类型，包括：`telegram` `feishu` `both` （当选择 `both` 后，同时发送到 Telegram 和飞书群）
  - `FEISHU_TOKEN` → 飞书 Bot webhook 地址
  - `SIMPREAD_ID` → 你在开放平台上设置的 ID
  - `SIMPREAD_TOKEN` → 你在开放平台上的 Token
  - `TELEGRAM_CHAT` → 前面得到的 Chat ID
  
  > 设置完毕后，应该类似下图所示：
  
  ![image-20220208112357996](https://cdn.jsdelivr.net/gh/23784148/upload-images@main/typora/20220208_1644290638.png)

- 设置并激活你的 Actions，进入 Actions，会看到下图所示的界面，并点击箭头所示的内容

  ![image-20220208113211574](https://cdn.jsdelivr.net/gh/23784148/upload-images@main/typora/20220208_1644291131.png)

  > 按下图依次点击箭头所示的内容，激活你的 Workflow

  ![image-20220208113322200](https://cdn.jsdelivr.net/gh/23784148/upload-images@main/typora/20220208_1644291202.png)

- 根据下图红色箭头所示依次点击，最后找到截图的 `send-daily.yml` 点击绿色箭头编辑它。

  ![image-20220208113631201](https://cdn.jsdelivr.net/gh/23784148/upload-images@main/typora/20220208_1644291391.png)

- 编辑红框中的内容，当确认无误后，依次点击红色箭头。

  ![image-20220209121332097](https://cdn.jsdelivr.net/gh/23784148/upload-images@main/typora/20220209_1644380012.png)

  > 注意：

  - `40` → **40分钟** 的意思
  - `11` → 国际标注时间，换算成北京时间就是 **19点**，如何换算可以 [看这里](http://www.timebie.com/cn/universalbeijing.php)。
  - 红色框的默认值是：每天晚上 **19点40分** 触发。
  - 你可以将 `40` 改为任意值，因为要设置为 **19点10分** 发送的话，性能会出现问题，所以改为 40

  点击上图所示的 **Commit changes** 后会自动触发 `send-daily.yml` 中的代码。

- 进入 Actions 会看到下面视频所示的效果，如果上述配置无误的话，会将每日回顾发送通过  [简悦 · 同步助手 小二](https://t.me/simpread_bot) 发送给你。

至此，全部配置结束。

# 视频效果

> 完成上述操作后的视频效果

https://cdn.jsdelivr.net/gh/23784148/upload-images@main/typora/2022-02-08_11-47-23.mp4

# 注意事项

- 上述教程以 Telegram 为例，如果使用飞书群与此方式类似，但无需 `TELEGRAM_CHAT`，但需要获得 `FEISHU_TOKEN` 。

-  `send-daily.yml` 的触发条件有两个：
  - 每天晚上 **19点40分** （你可以根据上图修改为任意时间）
  - 每次 Push 代码到 main 分支（也就是当你修改代码并 push 后自动执行）

- 每次发送的每日回顾内容可以通过**简悦扩展端 → 稍后读 → 24小时**查看。（如果你配置完没有收到的话，看看这里是否存在稍后读，如果没有的话，则不会发送。

  ![image-20220208115809660](https://cdn.jsdelivr.net/gh/23784148/upload-images@main/typora/20220208_1644292689.png)

# 进阶配置

上述配置使用了  [简悦 · 同步助手 小二](https://t.me/simpread_bot) 如果你有自己的 Channel 或群，可以自己建立一个 bot，然后增加一个新的 New repository secret  `TELEGRAM_TOKEN` 并设置为这个 bot token。

![image-20220209121730858](https://cdn.jsdelivr.net/gh/23784148/upload-images@main/typora/20220209_1644380250.png)

同时将这个 telegram bot 加入到你需要的 Channel 或者 Group，并设置为管理员权限，仍旧通过  [@username_to_id_bot](https://t.me/username_to_id_bot) 得到 Channel 或者 Group 的 Chat ID。

最后，简悦的每日回顾就会推送到你的  Channel 或者 Group。

# 关联

- [适配飞书妙记](https://github.com/Kenshin/simpread/discussions/3190)
- [发送当前 URL 到飞书群（包括：加入稍后读或手动触发）](https://github.com/Kenshin/simpread/discussions/3104)
- [将当前阅读模式转换为 Telegraph Page 并发送到你的 Telegram Channel](https://github.com/Kenshin/simpread/discussions/3337)
- [适配了一组关于 Telegram 的新站](https://github.com/Kenshin/simpread/discussions/2883)
- [保存到 Telegraph](https://github.com/Kenshin/simpread/discussions/2795)
- [如何玩转 Telegram bot（通过 Telegram bot 收藏到简悦的稍后读）](https://github.com/Kenshin/simpread/discussions/2792)

# 附录

>  如何获取每次来自 https://github.com/Kenshin/simpread-actions 最新的代码。

当你 Fork 代码后，如果 [SimpRead Actions](https://github.com/Kenshin/simpread-actions) 没更新， **Fetch and Merge** 无法点击。

![image-20220209123044486](https://cdn.jsdelivr.net/gh/23784148/upload-images@main/typora/20220209_1644381044.png)

反之，如果 [SimpRead Actions](https://github.com/Kenshin/simpread-actions) 更新了会出现下图效果。

![image-20220209122718241](https://cdn.jsdelivr.net/gh/23784148/upload-images@main/typora/20220209_1644380838.png)

点击 **Fetch and Merge**，稍等片刻后就会将最新的代码 Merge 到你的 repo，同时也马上推送每日回顾（如果有的话）到你的 Telegram。

**注意：**  一般来说如果只改动 `send-daily.yml` 里面的时间（也就是下图波浪线所示的内容）的话，上述操作可 **自动 Merge**

![image-20220209123608442](https://cdn.jsdelivr.net/gh/23784148/upload-images@main/typora/20220209_1644381368.png)

**Comment by twi1ight00** [comment]
请教大佬，我试了一下，action已经触发，但是响应是404：
![image](https://user-images.githubusercontent.com/22125281/183801895-3fc9ba12-8262-4478-9151-f0227e44f4f9.png)
![image](https://user-images.githubusercontent.com/22125281/183807057-bdafebf3-1ed0-4d38-934d-3db356769f3a.png)

讨论 https://github.com/Kenshin/simpread/issues/4354
